openrtb_ext/request_wrapper.go

```diff
@@ -31,46 +31,51 @@ import (
 // NOTE: The RequestWrapper methods (particularly the ones calling (un)Marshal are not thread safe)
 type RequestWrapper struct {
 	*openrtb2.BidRequest
-	imp         []*ImpWrapper
-	impAccessed bool
-	userExt     *UserExt
-	deviceExt   *DeviceExt
-	requestExt  *RequestExt
-	appExt      *AppExt
-	regExt      *RegExt
-	siteExt     *SiteExt
-	sourceExt   *SourceExt
+	impWrappers         []*ImpWrapper
+	impWrappersAccessed bool
+	userExt             *UserExt
+	deviceExt           *DeviceExt
+	requestExt          *RequestExt
+	appExt              *AppExt
+	regExt              *RegExt
+	siteExt             *SiteExt
+	sourceExt           *SourceExt
 }
 
 const jsonEmptyObjectLength = 2
 
 // LenImp returns the number of impressions without causing the creation of ImpWrapper objects.
 func (rw *RequestWrapper) LenImp() int {
-	if rw.imp == nil {
-		return len(rw.Imp)
+	if rw.impWrappersAccessed {
+		return len(rw.impWrappers)
 	}
 
-	return len(rw.imp)
+	return len(rw.Imp)
 }
 
 func (rw *RequestWrapper) GetImp() []*ImpWrapper {
```

Comment:
I went back and forth on this one. I decided to name it `GetImp` to match the pattern used for all other wrappers. Like `GetUser` returns a wrapper for the `User` field, `GetImp` returns a wrapper for the `Imp` field. I agree the name is confusing and choose to match the existing pattern. Looking for other opinions on how to name this here.

Reply:
I think I'm ok with `GetImp`. I too was a bit confused by the naming when reading the code where it's called and some of that confusion can be attributed to the naming plurality and some to the fact it returns a wrapper instead of an `Imp`. However I tend to favor the consistency in how the `Get` and `Set` functions are named so I think `GetImp` works.